pkg/odo/cli/component/list.go

```diff
@@ -104,57 +147,126 @@ func (lo *ListOptions) Run() (err error) {
 		}
 		return nil
 	}
-	var components component.ComponentList
 
-	if lo.allAppsFlag {
-		// retrieve list of application
-		apps, err := application.List(lo.Client)
+	// non --path workflow below
+	// read the code like
+	// -> experimental
+	//	or	|-> --all-apps
+	//		|-> the current app
+	// -> non-experimental
+	//	or	|-> --all-apps
+	//		|-> the current app
+
+	// experimental workflow
+
+	if lo.isExperimentalMode && util.CheckPathExists(lo.devfilePath) {
+
+		var deploymentList *appsv1.DeploymentList
+		var err error
+
+		// TODO: wrap this into a component list for docker support
+		if lo.allAppsFlag {
+			deploymentList, err = lo.KClient.ListAllDeployments()
+
+		} else {
+			deploymentList, err = lo.KClient.ListDeployments(lo.Application)
+		}
+
 		if err != nil {
 			return err
 		}
 
-		var componentList []component.Component
+		envinfo := lo.EnvSpecificInfo.EnvInfo
+		currentComponentState := "Unpushed"
+		currentComponentName := envinfo.GetName()
+		lo.hasDevfileComponents = true
+		w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+		fmt.Fprintln(w, "Devfile Components: ")
+		fmt.Fprintln(w, "APP", "\t", "NAME", "\t", "PROJECT", "\t", "TYPE", "\t", "STATE")
+		for _, comp := range deploymentList.Items {
+			app := comp.Labels[applabels.ApplicationLabel]
+			cmpType := comp.Labels[componentlabels.ComponentTypeLabel]
+			if comp.Name == currentComponentName && app == lo.Application && comp.Namespace == envinfo.GetNamespace() {
+				currentComponentState = "Pushed"
```

Comment:
Can we define `Pushed` & `Unpushed` as consts?

Reply:
done